src/user/mod.rs

```diff
@@ -103,16 +102,18 @@ impl User {
     }
 
     /// Queries the database for a user with a certain `api_token` value.
-    pub fn find_by_api_token(conn: &GenericConnection, token: &str) -> CargoResult<User> {
-        let stmt = conn.prepare(
-            "SELECT * FROM users \
-                                      WHERE api_token = $1 LIMIT 1",
-        )?;
-        let rows = stmt.query(&[&token])?;
-        rows.iter()
-            .next()
-            .map(|r| Model::from_row(&r))
-            .chain_error(|| NotFound)
+    pub fn find_by_api_token(conn: &PgConnection, token_: &str) -> CargoResult<User> {
+        // TODO: This should be doable as a single query, but I can't figure
```

Comment:
What's the SQL that you would write to do this as a single query? I don't see any way that you could do this in one query.